src/client/js/LinkUrlDialog.js

```diff
@@ -16,33 +16,34 @@ export default class LinkUrlDialog extends React.Component {
         this.onCancelLink = this.onCancelLink.bind(this);
         this.onRemoveLink = this.onRemoveLink.bind(this);
 
-
-        console.log(this.props.titleValue);
-
         this.state = {
             titleValue: this.props.titleValue,
-            urlValue: '',
-            descValue: ''
+            urlValue: this.props.urlValue,
+            descValue: this.props.descValue
         };
+        console.log(this.props.titleValue);
 
     }
+
+    componentWillReceiveProps() {
+        this.state = {
+            titleValue: props.titleValue,
+            urlValue: props.url,
+            descValue: props.descValue
+        };
+    }
+
     onEditLink(e) {
         e.preventDefault();
-        this.props.onEditLink(e);
+        const {titleValue, urlValue, descValue} = this.state;
+        this.props.onEditLink(titleValue,urlValue,descValue);
     }
     onConfirmLink(e) {
-        e.preventDefault();
-        const {
-            titleValue, urlValue, descValue
-        } = this.state;
-        const entityKey = Entity.create('LINK', 'MUTABLE', {
-            linkTitle: titleValue,
-            linkUrl: urlValue,
-            linkDesc: descValue,
-            onLinkEdit: this.props.promptForEdit
-        });
-        this.props.onConfirmLink(entityKey);
```

Comment:
LinkUrlDialog should be decoupled from work with Editor, it shoud just return (titleValue,urlValue,descValue), that's all that needed from this